File: C Programming/Tower of Hanoi/towerOfHanoi.c
#include <stdio.h>

void towerOfHanoi(int n, char from_rod, char to_rod, char aux_rod){
	if (n == 0){
		return;
	}
	towerOfHanoi(n - 1, from_rod, aux_rod, to_rod);
	printf("Move disk %d from rod %c to rod %c\n", n, from_rod, to_rod);
	towerOfHanoi(n - 1, aux_rod, to_rod, from_rod);
}

int main()
{
	int disks;
	printf("Input the number of disks: ");
	scanf_s("%d", &disks);

	towerOfHanoi(disks, 'A', 'C', 'B');
	return 0;
}

File: Python Programming/Tower Of Hanoi/towerOfHanoi.py
# Tower of Hanoi solution using a recursive function in Python3
def TowerOfHanoi(n, from_rod, to_rod, aux_rod):
    if n == 0:
        return
    TowerOfHanoi(n-1, from_rod, aux_rod, to_rod)
    print("Move disk", n, "from rod", from_rod, "to rod", to_rod)
    TowerOfHanoi(n-1, aux_rod, to_rod, from_rod)


# Input prompt for the user to enter the number of disks
print("Enter the number of disks: ")
disks = int(input())

# Function call
TowerOfHanoi(disks, 'A', 'C', 'B')
